### DISCUSSION TITLE
Using hooks inside 'createBrowserRouter' 'loader'

### DISCUSSION THREAD

**Discussion by ralph993** [remix-run/react-router#9246] [Q&A] [answered]
I'm trying to use react-query  inside the loader function using createBrowserRouter.

So, what I did was this: 
```js
const router = createBrowserRouter([
{
	path: "/",
	element: <Home />,
	loader:
		async ({ _, params }) =>
		() => {
			const { data, isLoading } = useFetchRoute(params);
			if (!isLoading && !data) throw new Response("Not Found", { status: 404 });
			return data;
		},
	}
]);
```

Is this considered bad practice?

**Comment by tkapitein** [comment, marked as answer]
It is not possible to use hooks inside loaders. This [article](https://tkdodo.eu/blog/react-query-meets-react-router) could help you integrating with React Query.

**Reply by ralph993** [reply]
Well, what i did worked, i just wanted to know if that was a bad practice. But it actually worked!

**Comment by bvjebin** [comment]
I have a similar requirement.  I want to get the auth state from redux in the loader to make some decision. As of now I don't find any way to inject that data into the loader. Any help would be appreciated.

**Reply by MrBista** [reply]
I have this problem too right now, is there solution that you found ?
I would be appreciate if there is way to solve the problem by your way

**Comment by vladifel** [comment]
I have a similar requirement. Trying to execute some Redux logic inside the loader.

**Comment by clementcreusat** [comment]
@ralph993 hello !!

Quick question, how did you type `useLoaderData` in your component ? 
Your are using curry function and typescript doesn't like `const data = useLoaderData()()`.

Thanks :)

**Reply by ralph993** [reply]
yeah, I wasn't using typescript in this project back then, I guess you can extend useLoaderData type so it can accept a Promise.
Keep in mind this is a hack. 

Hope this help.
